### ISSUE TITLE
"Android: Install SDK & Create Keys" fails.

### ISSUE THREAD

**Issue by Casteele72** [renpy/renpy#3317]
While trying out Ren'Py, I tried to build for Android, which required "Install SDK & Create Keys". However, my initial attempt failed and gave a link to install a usable Java SDK (from Adoptium). I followed and installed that Java SDK because Ren'Py did not seem to accept any Java SDK installed on my system (I have OpenJDK, Bellsoft, and Oracle installed side-by-side, with OpenJDK as the default).

I finally got the Java SDK from Adopium installed and working (I also had to edit "renpy.sh" to add a "JAVA_HOME" environment variable), and received the following error:

> Ren'Py 7.4.11.2266
> 
> I'm compiling a short test program, to see if you have a working JDK on your system.
> 
> 
> 
> Traceback (most recent call last):
>   File "game/mobilebuild.rpy", line 195, in call
>   File "/home/tom/ab/renpy-build/tmp/install.linux-x86_64/lib/python2.7/subprocess.py", line 325, in \_\_init\_\_
>   File "/home/tom/ab/renpy-build/tmp/install.linux-x86_64/lib/python2.7/subprocess.py", line 978, in _execute_child
> OSError: [Errno 2] No such file or directory
> 
> I was unable to use javac to compile a test file. If you haven't installed the Java Development Kit yet, please download it from:
> 
> {a=https://adoptium.net/?variant=openjdk8}https://adoptium.net/?variant=openjdk8{/a}
> 
> The JDK is different from the JRE, so it's possible you have Java without having the JDK. Please make sure you installed the 'JavaSoft (Oracle) registry keys'.
> 
> Without a working JDK, I can't continue.

That does not appear to be Java-related, but a Ren'Py (and/or Python) script and path issue: there is no "/home/tom" on my system.

I cannot try to fix this or proceed further at this point since I am not familiar enough with Ren'Py to locate and fix the issue.

Edit to add:

My operating system is Debian Linux 10.
The Ren'Py version I am working with is from the Ren'P website, https://www.renpy.org/dl/7.4.11/renpy-7.4.11-sdk.zip

**Comment by renpytom** [comment]
My guess is that JAVA_HOME being set is a mistake - please unset it. 

What Ren'Py is trying to do in this test is to compile renpy-7.4.11/rapt/buildlib/CheckJDK8.java. 

What I'd suggest trying is:

cd ~/renpy-7.4.11/rapt/buildlib/
javac CheckJDK8.java
java CheckJDK8 || echo Error.

And figure out which command fails.

**Comment by Casteele72** [comment]
"JAVA_HOME" was not a mistake. Without it:
1) The build process fails stating that it cannot find Java;
2) The link given to install a compatible Java SDK suggested setting it to override my default system installed Java SDK.

Here are the results requested (adjusted for my path configuration, and added the default system Java SDK setting for "JAVA_HOME" via echo):

~/devel/renpy-7.4.11-sdk/rapt$ echo $JAVA_HOME
/usr/lib/jvm/java-16-oracle
~/devel/renpy/renpy-7.4.11-sdk/rapt/buildlib$ javac CheckJDK8.java
~/devel/renpy-7.4.11-sdk/rapt/buildlib$ java CheckJDK8 || echo Error.
Error.

I get the same results when I change JAVA_HOME to point to the Adoptium Java SDK (which is "/opt/adoptium/openjdk/temurin/jdk8u322-b06/bin", from the Adoptium install instructions).

To give a little more detail on Adoptium...
I installed the version obtain by following the link Ren'Py gave me ( https://adoptium.net/?variant=openjdk8 ), and following their instructions ( https://adoptium.net/installation.html?variant=openjdk8#x64_linux-jdk ). The version that was downloaded and installed is https://github.com/adoptium/temurin8-binaries/releases/download/jdk8u322-b06/OpenJDK8U-jdk_x64_linux_hotspot_8u322b06.tar.gz

Would it make any difference that I installed the 64-bit version and not the 32-bit version (if there even is one... I did not check)?

I also noticed that in m system install, the .../bin directory was not specified, but the Adoptium install does include it in the path. But it seems to make no difference if I omit it on those commands... It still errors.

HOWEVER...

I changed the renpy.sh script to omit the tail .../bin from the JAVA_HOME, and that seems to have worked. It is now downloading the Android SDK as I write this. (Another note is that I already have a *working* Android SDK installed in my system for developing pure C/C++ Android apps...)
